openforcefield/topology/molecule.py

```diff
@@ -2194,11 +2200,71 @@ def to_networkx(self):
         for bond in self.bonds:
             G.add_edge(
                 bond.atom1_index, bond.atom2_index, bond_order=bond.bond_order, is_aromatic=bond.is_aromatic,
-                stereochemistry = bond.stereochemistry)
+                stereochemistry=bond.stereochemistry)
             #G.add_edge(bond.atom1_index, bond.atom2_index, attr_dict={'order':bond.bond_order})
 
         return G
 
+    def find_rotatable_bonds(self, ignore_functional_groups=None, toolkit_registry=GLOBAL_TOOLKIT_REGISTRY):
+        """
+        Find all bonds classed as rotatable ignoring any matched to the ``ignore_functional_groups`` list.
+
+        Parameters
+        ----------
+        ignore_functional_groups: optional, List[str], default=None,
+            A list of bond SMARTS patterns to be ignored when finding rotatable bonds.
+
+        toolkit_registry: openforcefield.utils.toolkits.ToolRegistry or openforcefield.utils.toolkits.ToolkitWrapper, optional, default=None
+            :class:`ToolkitRegistry` or :class:`ToolkitWrapper` to use for SMARTS matching
+
+        Returns
+        -------
+        bonds: List[openforcefield.topology.molecule.Bond]
+            The list of openforcefield.topology.molecule.Bond instances which are rotatable.
+        """
+
+        # general rotatable bond smarts taken from RDKit
+        # https://github.com/rdkit/rdkit/blob/1bf6ef3d65f5c7b06b56862b3fb9116a3839b229/rdkit/Chem/Lipinski.py#L47%3E
+        rotatable_bond_smarts = '[!$(*#*)&!D1:1]-&!@[!$(*#*)&!D1:2]'
+
+        # get all of the general matches
+        general_matches = self.chemical_environment_matches(query=rotatable_bond_smarts,
+                                                            toolkit_registry=toolkit_registry)
+
+        # this will give all forwards and backwards matches, so condense them down with this function
+        def condense_matches(matches):
+            condensed_matches = set()
+            for m in matches:
+                condensed_matches.add(tuple(sorted(m)))
+            return condensed_matches
+
+        general_bonds = condense_matches(general_matches)
+
+        # now refine the list using the ignore groups
+        if ignore_functional_groups is not None:
+            matches_to_ignore = set()
+            if not isinstance(ignore_functional_groups, list):
+                ignore_functional_groups = [ignore_functional_groups]
```

Comment:
Since there are iterable objects other than lists, we try to follow a design pattern like this: 

https://github.com/openforcefield/openforcefield/blob/ee0f715e094e6f1749bf60f3a3b48174a9e6a9b7/openforcefield/topology/topology.py#L1030-L1035)

Reply:
Thanks for catching that I have made that change.